### DISCUSSION TITLE
NoSQL Revolution | 夜天之书

### DISCUSSION THREAD

**Discussion by giscus** [tisonkun/blog#27] [Comments]
# NoSQL Revolution | 夜天之书

从本世纪初谷歌的三篇论文发布以来，数据处理领域在大数据的方向上探索了将近二十年的时间。从三篇论文的开源实现 Apache Hadoop 和 Apache HBase 开始，到打破传统关系型数据库的分布式数据处理系统如雨后春笋般接连诞生，NoSQL 系统回应了移动互联时代的数据爆发式增长的挑战。 诚然，传统的数据库专家对 NoSQL 也有像 MapReduce: A major step backw

https://www.tisonkun.org/2022/06/13/nosql-revolution/

**Comment by mapleFU** [comment]
我感觉很多业务系统，都相当于自己在 "NoSQL" 上实现了一遍事务。要么是把某个信息源当成类似 2PC 里面 Coordinator 的东西，认为这上面的才是唯一靠谱的最终结果；要么把分布式事务拆成业务的（比如 TCC/SAGA 这种）。我也比较好奇，业务一般什么时候会用到数据库提供的分布式事务呢（我看 NoSQL 好像就 MongoDB 做了基于 Clock-SI 的分布式事务，但是不知道用户使用体验怎么样）

**Reply by tisonkun** [reply]
业务在发现 NoSQL 数据库的“事务”能解决自己要的语义的时候就会用。比如 HBase 单行事务对某些场景也够用，Cassandra 可以选择强一致性进行查询。消息队列里的“事务”能够原子地提交一批数据。Redis 的“事务”对于熟悉 Sagas 的人来说也不是不能用。

业务系统做分布式事务事务的不多，要么就是 Sagas 式的 ACD + 补偿的“事务”，要么就是分库分表保证对应数据在同一个物理数据库上依赖关系数据库事务。

业务把 ZK 或者 etcd 当成信源来用也不是没有，但是具体情况具体看待吧，我看到的一些 case 上面已经说了。

**Reply by mapleFU** [reply]
学到了，所以还是看有没有，没有感觉对大部分业务来说...好像也能忍，然后自己造
